resources/sdk/purecloudjava-guest/templates/Configuration.mustache

```diff
@@ -20,5 +34,227 @@ public class Configuration {
    */
   public static void setDefaultApiClient(ApiClient apiClient) {
     defaultApiClient = apiClient;
+    logger = defaultApiClient.getLogger();
+    configFilePath = defaultApiClient.getConfigFilePath();
+    loggingConfiguration = defaultApiClient.getLoggingConfiguration();
+    Configuration configuration = new Configuration();
+    configuration.applyConfigFromFile();
+    configuration.configureAutoReload();
+  }
+
+  private boolean configureAutoReload() {
+    boolean shouldUnsubscribeFromFileWatcher = !defaultApiClient.getAutoReloadConfig()
+            && configFileWatcher != null
+            && configFileWatcher.getListeners().contains(configListener);
+    boolean shouldSubscribeToFileWatcher = defaultApiClient.getAutoReloadConfig()
+            && (configFileWatcher == null || configFileWatcher.getListeners().size() == 0);
+    if (shouldUnsubscribeFromFileWatcher) {
+      return true;
+    }
+    if (shouldSubscribeToFileWatcher) {
+      if (configFileWatcher == null) {
+        initializeFileWatcher();
+      } else {
+        configFileWatcher.addListener(configListener);
+      }
+    }
+    return false;
+  }
+
+  private void initializeFileWatcher() {
+    File parentFolder = new File(configFilePath).getParentFile();
+    Path configFileName = Paths.get(configFilePath).getFileName();
+    while (parentFolder != null && !parentFolder.exists()) {
+      File temp = parentFolder.getParentFile();
+      if (temp != null) {
+        parentFolder = temp;
+      } else {
+        break;
+      }
+    }
+    if (parentFolder != null) {
+      configListener = new FileWatcher.FileAdapter() {
+        public void onModified(FileWatcher.FileEvent event) {
```

Comment:
I'm getting a build error saying onModified and onCreated should return booleans

Reply:
Fixed this